src/main/drivers/accgyro/accgyro_spi_bmi160.c

```diff
@@ -377,8 +379,8 @@ static bool bmi160GyroRead(gyroDev_t *gyro)
                 dev->txBuf[1] = BMI160_REG_GYR_DATA_X_LSB | 0x80;
                 gyro->segments[0].len = 14;
                 gyro->segments[0].callback = bmi160Intcallback;
-                gyro->segments[0].u.buffers.txData = dev.txBuf[1];
-                gyro->segments[0].u.buffers.rxData = dev.rxBuf[1];
+                gyro->segments[0].u.buffers.txData = &dev->txBuf[1];
+                gyro->segments[0].u.buffers.rxData = &dev->rxBuf[1];
```

Comment:
I am not 100% sure but can you please recheck that it is working on some board with BMI160 and that no memory overflow could happen?

I compared this with BMI270 driver and this starts at `&dev->txBuf[1]` which is in fact `dev->txBuf` and also `gyro->segments[0].len = 14;`. We are starting at 1 and 14. This line `dev->txBuf[1] = BMI160_REG_GYR_DATA_X_LSB | 0x80;` also uses `dev->txBuf[0] = BMI270_REG_GYR_DATA_X_LSB | 0x80;`. Maybe I don't understand code maybe exactly this is the difference between BMI160 and BMI270. Just asking.

Reply:
Thanks @daleckystepan

The read data on the BMI270 has an extra byte of padding so the 16 bit data can be read but casting rxBuf to (int16_t *) and reading element [1]. The BMI160 lacks this padding. See https://github.com/betaflight/betaflight/pull/11568#issuecomment-1130381018.

In reviewing the code again I see that I was casting the raw data to (uint16_t *) instead of (int16_t *) so I've just pushed a change to fix that. It worked, but this is now cleaner.

Regarding the length of the transfer on the BMI270 we have the register address being written, plus a byte of padding, then 6 bytes of acc data followed by 6 bytes of gyro data, so 14 bytes in total.

On the BMI160 we have the register address, no padding, then 6 bytes of acc data followed by 6 bytes of gyro data, so as you point out, only 13 bytes need to be read. Reading the 14th byte wasn't doing any harm besides wasting a little time.

Reply:
Thank you for looking at it. Approved.